pyqtgraph/parametertree/parameterTypes.py

```diff
@@ -741,3 +748,635 @@ class TextParameter(Parameter):
+        fList = fileDlg.selectedFiles()
+    else:
+        fList = []
+    if relativeTo is not None:
+        # Forward slashes are returned by qt, so preserve this convention regardless
+        # of the os. Since os.path will always prefer os-flavored separators, use pathlib to consistently
+        # return posix flavor
+        newFileList = []
+        for file in fList:
+            # pathlib and darwin have case sensitivity issues using relative_to, so use os.path for this
+            relative = os.path.relpath(file, relativeTo)
+            relativePosix = Path(relative).as_posix()
+            newFileList.append(relativePosix)
+        fList = newFileList
+    if fileDlg.fileMode() == fileDlg.FileMode.ExistingFiles:
+        return fList
+    elif len(fList) > 0:
+        return fList[0]
+    else:
+        return None
+
+class FileParameterItem(WidgetParameterItem):
+    def __init__(self, param, depth):
+        self._value = None
+        # Temporarily consider string during construction
+        oldType = param.opts.get('type')
+        param.opts['type'] = 'str'
+        super().__init__(param, depth)
+        param.opts['type'] = oldType
+
+        button = QtWidgets.QPushButton('...')
+        button.setFixedWidth(25)
+        button.setContentsMargins(0, 0, 0, 0)
+        button.clicked.connect(self._retrieveFileSelection_gui)
+        self.layoutWidget.layout().insertWidget(2, button)
+        self.displayLabel.resizeEvent = self._newResizeEvent
+        # self.layoutWidget.layout().insertWidget(3, self.defaultBtn)
+
+    def makeWidget(self):
+        w = super().makeWidget()
+        w.setValue = self.setValue
+        w.value = self.value
+        # Doesn't make much sense to have a 'changing' signal since filepaths should be complete before value
+        # is emitted
+        delattr(w, 'sigChanging')
+        return w
+
+    def _newResizeEvent(self, ev):
+        ret = type(self.displayLabel).resizeEvent(self.displayLabel, ev)
+        self.updateDisplayLabel()
+        return ret
+
+    def setValue(self, value):
+        self._value = value
+        self.widget.setText(asUnicode(value))
+
+    def value(self):
+        return self._value
+
+    def _retrieveFileSelection_gui(self):
+        curVal = self.param.value()
+        if isinstance(curVal, list) and len(curVal):
+            # All files should be from the same directory, in principle
+            # Since no mechanism exists for preselecting multiple, the most sensible
+            # thing is to select nothing in the preview dialog
+            curVal = Path(curVal[0])
+            if curVal.is_file():
+                curVal = curVal.parent
+        useDir = curVal or str(os.getcwd())
+        opts = self.param.opts
+        startDir = Path(useDir).absolute()
+        if startDir.is_file():
+            opts['selectFile'] = startDir.name
+            startDir = startDir.parent
+        if startDir.exists():
+            opts['directory'] = str(startDir)
```

Comment:
wrapping `str(Path)` can cause huge problems, if you decide to stick with pathlib, use Path.as_posix(), or for a path-or-string compatible method use `os.fsdecode(startDir)`

Reply:
Helpful feedback, thanks. I wish these tidbits were easier to find while perusing the internet (first example coming to mind is this should be in the pathlib docs!)

Reply:
The official examples even state if you need a string version, just use `str()`